lib/aurora_bootstrapper/notifier.rb

```diff
@@ -12,7 +12,59 @@ def export_date
     
     # ENV is string to string dictionary
     def export_date_override
-      DateTime.now.strftime("%Y-%m-%d") if ENV.key?('EXPORT_DATE_OVERRIDE')
+      datetime_subfolder_str = nil
+
+      if ENV.key?('EXPORT_DATE_OVERRIDE')
+        now = Date.today
+        
+        # expiration time is 30 days
+        for i in 1..30
+          datetime_subfolder_str = check_db_dump_done_for_one_day( datetime: (now-i))
+          unless datetime_subfolder_str.nil?
+            break
+          end
+        end
+
+        # the first run for the given db partition
+        if datetime_subfolder_str.nil?
+          datetime_subfolder_str = now.strftime("%Y-%m-%d")
+        end
+      end
+
+      datetime_subfolder_str
+    end
+
+    def check_db_dump_done_for_one_day( datetime: )
+      datetime_subfolder_str = nil
+      prefix = [ bucket_path, datetime.strftime("%Y-%m-%d") ].join( '/' )
+
+      resp = client.list_objects_v2({
+        bucket: bucket,
+        prefix: prefix
+      })
+
+      if parse_resp(resp: resp)
```

Comment:
The [convention](https://blog.codeminer42.com/and-understanding-one-of-rubys-coolest-naming-conventions-5a9300b75605/) in Ruby is to name predicate functions as a question.

Reply:
I see